### PR TITLE
feat(hpc-vmware-managed-vcd): add pcc-migration condition

### PR DESCRIPTION
**Pull request by chipp972**
ref: MANAGER-15398

| Question         | Answer
| ---------------- | ---
| Branch?          | `develop` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-15398 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/details/dedicatedCloud.routing.js [hunk 1]
```diff
@@ -95,9 +95,12 @@ export default /* @ngInject */ ($stateProvider) => {
...
         ovhFeatureFlipping
-          .checkFeatureAvailability('hpc-vmware-managed-vcd')
-          .then((featureAvailability) =>
-            featureAvailability.isFeatureAvailable('hpc-vmware-managed-vcd'),
+          .checkFeatureAvailability(['hpc-vmware-managed-vcd', 'pcc-migration'])
+          .then(
+            (featureAvailability) =>
+              featureAvailability.isFeatureAvailable(
+                'hpc-vmware-managed-vcd',
+              ) && featureAvailability.isFeatureAvailable('pcc-migration'),
```
Comment (darsene) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/details/dedicatedCloud.routing.js:103]: As it's a sub feature, it should be like that rather
```suggestion
          .checkFeatureAvailability('hpc-vmware-managed-vcd:pcc-migration')
          .then(
            (featureAvailability) =>
               featureAvailability.isFeatureAvailable(hpc-vmware-managed-vcd:pcc-migration')),
```

File: packages/manager/apps/dedicated/client/app/managedBaremetal/details/managed-baremetal.routing.js [hunk 2]
```diff
@@ -93,9 +93,12 @@ export default /* @ngInject */ ($stateProvider) => {
...
         ovhFeatureFlipping
-          .checkFeatureAvailability('hpc-vmware-managed-vcd')
-          .then((featureAvailability) =>
-            featureAvailability.isFeatureAvailable('hpc-vmware-managed-vcd'),
+          .checkFeatureAvailability(['hpc-vmware-managed-vcd', 'pcc-migration'])
+          .then(
+            (featureAvailability) =>
+              featureAvailability.isFeatureAvailable(
+                'hpc-vmware-managed-vcd',
+              ) && featureAvailability.isFeatureAvailable('pcc-migration'),
```
Comment (darsene) [inline packages/manager/apps/dedicated/client/app/managedBaremetal/details/managed-baremetal.routing.js:101]:
```suggestion
          .checkFeatureAvailability('hpc-vmware-managed-vcd:pcc-migration')
          .then(
            (featureAvailability) =>
              featureAvailability.isFeatureAvailable(
                'hpc-vmware-managed-vcd:pcc-migration',
              ) ,
```

Comment (chipp972) [reply packages/manager/apps/dedicated/client/app/managedBaremetal/details/managed-baremetal.routing.js]: done

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/details/dedicatedCloud.routing.js (hunk 1 above)
Comment (chipp972) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/details/dedicatedCloud.routing.js]: done

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/details/dedicatedCloud.routing.js [hunk 3]
```diff
@@ -95,9 +95,11 @@ export default /* @ngInject */ ($stateProvider) => {
 
       managedVCDAvailability: /* @ngInject */ (ovhFeatureFlipping) =>
         ovhFeatureFlipping
-          .checkFeatureAvailability('hpc-vmware-managed-vcd')
+          .checkFeatureAvailability(['hpc-vmware-managed-vcd:pcc-migration'])
```
Comment (pauldkn) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/details/dedicatedCloud.routing.js:98]: Are brackets `[ ]` still needed ?

**Review by pauldkn — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/details/dedicatedCloud.routing.js (hunk 3 above)
Comment (chipp972) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/details/dedicatedCloud.routing.js]: fixed

**Pull request merged**